src/folly/FollyKeepaliveTimer.cpp

```diff
@@ -8,7 +8,7 @@ FollyKeepaliveTimer::FollyKeepaliveTimer(
     folly::EventBase& eventBase,
     std::chrono::milliseconds period)
     : eventBase_(eventBase),
-      running_(std::make_shared<bool>(false)),
+      state_(std::make_shared<uint32_t>(0)),
```

Comment:
do you want to rename state_ to generation_ and initialize the member variable on line 33?